### PR TITLE
DOCS: Lightspeed example

### PR DESCRIPTION
**Pull request by springcoil**
Added lightspeed example based on the Stan docs. 
Slight alterations were needed to get it to work.

### CODE REVIEW

File: pymc3/examples/lightspeed_example.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    #mu = pm.Uniform('mu', lower = -tt.inf, upper= np.inf)
+    #sigma = pm.Uniform('sigma', lower = 0, upper= np.inf)
+
+
+    #using vague priors works
+    mu = pm.Uniform('mu', lower = light_speed.std()/1000.0, upper= light_speed.std()*1000.0)
+    sigma = pm.Uniform('sigma', lower = light_speed.std()/1000.0, upper= light_speed.std()*1000.0)
+
+    # define likelihood
+    y_obs = pm.Normal('Y_obs', mu = mu, sd = sigma, observed = light_speed)
```
Comment (twiecki) [inline pymc3/examples/lightspeed_example.py]: Can you make this all pep8? `mu=mu`, without the spaces (only when doing assignments).

Comment (springcoil) [inline pymc3/examples/lightspeed_example.py]: Fixed in the new version I think.

**Comment by springcoil** [comment]
I'm getting very similar errors to your error @olafSmits did you ever resolve it? Seems to be the for loop problem again - have you some code that fixes this?

**Comment by springcoil** [comment]
Hey @twiecki my commit with the GARCH model is here too.

**Comment by olafSmits** [comment]
No I didn't resolve it. I got it working on a Windows 7 machine (with theano properly installed), but not on my laptop (Ubuntu 14.04; theano installed but not working on GPU). It's indeed something with the for loop. Maybe it's worth replacing the loop by a `theano.scan` call? That requires some re-writing though.

**Comment by springcoil** [comment]
I had an attempt at writing theano.scan but I found it difficult. Can anyone like @twiecki help me refactor this? 
I think using shared variables http://deeplearning.net/software/theano/library/scan.html solves this problem - I'll get to it later :) 

I get the following error - see the next commit for the code. 

`(pymc3_examples)Peadars-MacBook-Pro% python pymc3/examples/garch_example.py
log_thunk_trace: There was a problem executing an Op.
Traceback (most recent call last):
  File "pymc3/examples/garch_example.py", line 47, in <module>
    beta_1=beta1, sigma_0=theta, observed=r)
  File "/Users/peadarcoyle/anaconda/envs/pymc3_examples/lib/python3.5/site-packages/pymc3-3.0-py3.5.egg/pymc3/distributions/distribution.py", line 25, in __new__
    return model.Var(name, dist, data)
  File "/Users/peadarcoyle/anaconda/envs/pymc3_examples/lib/python3.5/site-packages/pymc3-3.0-py3.5.egg/pymc3/model.py", line 256, in Var
    var = ObservedRV(name=name, data=data, distribution=dist, model=self)
  File "/Users/peadarcoyle/anaconda/envs/pymc3_examples/lib/python3.5/site-packages/pymc3-3.0-py3.5.egg/pymc3/model.py", line 520, in __init__
    self.logp_elemwiset = distribution.logp(data)
  File "/Users/peadarcoyle/anaconda/envs/pymc3_examples/lib/python3.5/site-packages/pymc3-3.0-py3.5.egg/pymc3/distributions/timeseries.py", line 101, in logp
    sigma = Normal.dist(mu=0., sd=x_prev/sigma).logp(x_i)
  File "/Users/peadarcoyle/anaconda/envs/pymc3_examples/lib/python3.5/site-packages/Theano-0.8.0.dev0-py3.5.egg/theano/tensor/var.py", line 203, in __truediv__
    return theano.tensor.basic.true_div(self, other)
  File "/Users/peadarcoyle/anaconda/envs/pymc3_examples/lib/python3.5/site-packages/Theano-0.8.0.dev0-py3.5.egg/theano/gof/op.py", line 668, in __call__
    required = thunk()
  File "/Users/peadarcoyle/anaconda/envs/pymc3_examples/lib/python3.5/site-packages/Theano-0.8.0.dev0-py3.5.egg/theano/gof/op.py", line 883, in rval
    fill_storage()
  File "/Users/peadarcoyle/anaconda/envs/pymc3_examples/lib/python3.5/site-packages/Theano-0.8.0.dev0-py3.5.egg/theano/gof/cc.py", line 1714, in __call__
    reraise(exc_type, exc_value, exc_trace)
  File "/Users/peadarcoyle/anaconda/envs/pymc3_examples/lib/python3.5/site-packages/six.py", line 686, in reraise
    raise value
ValueError: Input dimension mis-match. (input[0].shape[0] = 7, input[1].shape[0] = 8)
(pymc3_examples)Peadars-MacBook-Pro%`

**Comment by springcoil** [comment]
Maybe @fonnesbeck you could help me fix that error, it looks quite messy to me.

**Comment by fonnesbeck** [comment]
Which loop are you talking about? I don't see one.

**Comment by springcoil** [comment]
Trying to port this - 

`

```
def logp(self, values):
    sigma = self.sigma_0
    alpha_0 = self.alpha_0
    alpha_1 = self.alpha_1
    beta_1 = self.beta_1

    x_prev = values[0]
    _logp = Normal.dist(0., sd=sigma).logp(x_prev)

    for x in values[1:]:
        sigma = pm.sqrt(alpha_0 + alpha_1 * (x_prev/sigma)**2
                        + beta_1 * sigma**2)
        _logp = _logp + pm.Normal(0., sd=sigma).logp(x)
        x_prev = x

    return _logp`
```

**Comment by twiecki** [comment]
Can that be vectorized like in `GaussianRandomWalk`?

**Comment by springcoil** [comment]
I tried to vectorize it but I get the following error - 

`+ 55e7bc2...a9ba665 lightspeed_example -> lightspeed_example (forced update)
(pymc3_examples)Peadars-MacBook-Pro% python pymc3/examples/garch_example.py
log_thunk_trace: There was a problem executing an Op.
Traceback (most recent call last):
  File "pymc3/examples/garch_example.py", line 47, in <module>
    beta_1=beta1, sigma_0=theta, observed=r)
  File "/Users/peadarcoyle/anaconda/envs/pymc3_examples/lib/python3.5/site-packages/pymc3-3.0-py3.5.egg/pymc3/distributions/distribution.py", line 25, in __new__
    return model.Var(name, dist, data)
  File "/Users/peadarcoyle/anaconda/envs/pymc3_examples/lib/python3.5/site-packages/pymc3-3.0-py3.5.egg/pymc3/model.py", line 256, in Var
    var = ObservedRV(name=name, data=data, distribution=dist, model=self)
  File "/Users/peadarcoyle/anaconda/envs/pymc3_examples/lib/python3.5/site-packages/pymc3-3.0-py3.5.egg/pymc3/model.py", line 520, in __init__
    self.logp_elemwiset = distribution.logp(data)
  File "/Users/peadarcoyle/anaconda/envs/pymc3_examples/lib/python3.5/site-packages/pymc3-3.0-py3.5.egg/pymc3/distributions/timeseries.py", line 102, in logp
    + beta_1 * sigma**2)
  File "/Users/peadarcoyle/anaconda/envs/pymc3_examples/lib/python3.5/site-packages/Theano-0.8.0.dev0-py3.5.egg/theano/tensor/var.py", line 203, in __truediv__
    return theano.tensor.basic.true_div(self, other)
  File "/Users/peadarcoyle/anaconda/envs/pymc3_examples/lib/python3.5/site-packages/Theano-0.8.0.dev0-py3.5.egg/theano/gof/op.py", line 668, in __call__
    required = thunk()
  File "/Users/peadarcoyle/anaconda/envs/pymc3_examples/lib/python3.5/site-packages/Theano-0.8.0.dev0-py3.5.egg/theano/gof/op.py", line 883, in rval
    fill_storage()
  File "/Users/peadarcoyle/anaconda/envs/pymc3_examples/lib/python3.5/site-packages/Theano-0.8.0.dev0-py3.5.egg/theano/gof/cc.py", line 1714, in __call__
    reraise(exc_type, exc_value, exc_trace)
  File "/Users/peadarcoyle/anaconda/envs/pymc3_examples/lib/python3.5/site-packages/six.py", line 686, in reraise
    raise value
ValueError: Input dimension mis-match. (input[0].shape[0] = 7, input[1].shape[0] = 8)`

**Comment by springcoil** [comment]
I created another PR because the class stuff was too complicated. If that passes you can merge that one

**Comment by springcoil** [comment]
Ok so the way I see this - i've successfully vectorized this. However I've not successfully managed the dimensions. I'll need to dive into this soon.

File: pymc3/distributions/timeseries.py [hunk 2]
```diff
@@ -71,3 +71,29 @@ def logp(self, x):
...
+        sigma = self.sigma_0
+        alpha_0 = self.alpha_0
+        alpha_1 = self.alpha_1
+        beta_1 = self.beta_1
+
+        x_im1 = x[:-1]
+        x_i = x[1:]
+        sigma = T.sqrt(alpha_0 + alpha_1 * (x_im1/sigma)**2
+                            + beta_1 * sigma**2)
+        innov_like = Normal.dist(0., beta_1=beta_1, sd=sigma).logp(x_i)
```
Comment (twiecki) [inline pymc3/distributions/timeseries.py:97]: `beta_1` is not a valid kwarg to `Normal` I think, did you mean `mu`?

Comment (springcoil) [inline pymc3/distributions/timeseries.py:97]: I changed that to mu - but I still get the same error.

Comment (fonnesbeck) [inline pymc3/distributions/timeseries.py:97]: There are still too many arguments. Should only be two.

Comment (springcoil) [inline pymc3/distributions/timeseries.py:97]: I have another bit of code to send in. Will do that.

**Comment by springcoil** [comment]
Since some of this commit was in https://github.com/pymc-devs/pymc3/pull/968 and some errors - I closed this PR and opened a new one that should work.

**Pull request closed without merging**